velox/docs/develop/connectors.rst

```diff
@@ -0,0 +1,91 @@
+     - Description
+   * - ConnectorSplit
+     - Basic unit that describes a chunk of data for processing.
+   * - DataSource
+     - Provides methods to consume and process a split. A DataSource can optionally consume a
+       dynamic filter during execution to prune some rows from the output vector.
+   * - DataSink
+     - Provides methods to write a Velox vector externally.
+   * - Connector
+     - Allows creating instances of a DataSource or a DataSink.
+   * - Connector Factory
+     - Enables creating instances of a particular connector.
+
+Velox currently has in-built support for the Hive Connector and the TPC-H Connector.
+Let's see how the above connector interfaces are implemented in the Hive Connector in detail below.
+
+Hive Connector
+--------------
+The Hive Connector can be used to read and write data files (Parquet, DWRF) residing on
+an external storage (S3, HDFS, GCS, Linux FS).
+
+HiveConnectorSplit
+~~~~~~~~~~~~~~~~~~
+The HiveConnectorSplit describes a data chunk using parameters including `file-path`,
+`file-format`, `start position`, `length`, `storage format`, etc..
+Given a set of Parquet files, users or applications are responsible for defining the splits.
+
+HiveDataSource
+~~~~~~~~~~~~~~
+The HiveDataSource implements the `addSplit` API that consumes a HiveConnectorSplit.
+It creates a file reader based on the file format, offset, and length. The supported file formats
+are DWRF and Parquet.
+The `next` API processes the split and returns a batch of rows. Users can continue to call
+`next` until all the rows in the split are fully read.
+HiveDataSource allows adding a dynamic filter using the `addDynamicFilter` API. This allows
+supporting :ref:`Dynamic Filter Pushdown<DynamicFilterPushdown>`.
+
+HiveDataSink
+~~~~~~~~~~~~
+The HiveDataSink writes vectors to files on disk. The supported file formats are DWRF and Parquet.
+The parameters to HiveDataSink also include column names, sorting, partitioning, and bucketing information.
+The `appendData` API instantiates a file writer based on the above parameters and writes a vector to disk.
+
+HiveConnector
+~~~~~~~~~~~~~
+The HiveConnector extends the `createDataSource` connector API to create instances of HiveDataSource.
+It also extends the `createDataSink` connector API to create instances of HiveDataSink.
+One of the parameters to these APIs is `ConnectorQueryCtx`, which provides means to specify a
+memory pool and connector configuration.
+
+HiveConnectorFactory
+~~~~~~~~~~~~~~~~~~~~
+The HiveConnectorFactory enabled creating instances of the HiveConnector. A `connector name` say "hive"
+is required to register the HiveConnectorFactory. Multiple instances of the HiveConnector can then be
+created by using the `newConnector` API by specifying a `connectorId` and connector config listed
+:doc:`here</configs>`.
+
+Storage Adapters
+~~~~~~~~~~~~~~~~
+The Hive Connector file reading and writing is supporting on a variety of distributed storage APIs.
```

Comment:
Hive Connector allows reading and writing files from a variety of distributed storage systems.